configmaptocrs/main.go

```diff
@@ -149,6 +156,65 @@ func decodeConfigFile(raw []byte) (*configFile, error) {
 	return cf, nil
 }
 
+// convertNamesToK8S gets a configFile object and converts all names
+// in it to names compatible with K8S resources, if necessary.
+func convertNamesToK8S(cf *configFile) error {
+	var err error
+
+	for i := 0; i < len(cf.Pools); i++ {
+		cf.Pools[i].Name, err = formatToK8S(cf.Pools[i].Name, "IpAddressPool")
```

Comment:
Can we maybe do a simple one-liner regex check on the name to see if it's invalid format before calling it straight away?
i.e. if it's valid continue to next iteration, else it will invoke the formatToK8S